plugins/out_skywalking/skywalking.c

```diff
@@ -0,0 +1,387 @@
+                        void *out_context, struct flb_config *config) {
+    int flush_ret = -1;
+    int tmp_ret = -1;
+    struct flb_output_sw *ctx = out_context;
+    struct flb_upstream_conn *conn = NULL;
+    struct flb_http_client *client = NULL;
+    void* buf;
+    size_t buf_len;
+    size_t sent_size;
+
+    conn = flb_upstream_conn_get(ctx->u);
+    if (!conn) {
+        flb_plg_error(ctx->ins, "failed to establish connection to %s:%i", ctx->host, ctx->port);
+        flush_ret = FLB_RETRY;
+        goto done;
+    }
+
+    tmp_ret =  sw_format(ctx, data, bytes, &buf, &buf_len);
+    if (tmp_ret != 0) {
+        flb_plg_error(ctx->ins, "failed to create buffer");
+        printf("err\n");
+        flush_ret = FLB_ERROR;
+        goto done;
+    }
+
+    client = flb_http_client(conn, FLB_HTTP_POST, ctx->uri,
+            (const char*)buf, buf_len, ctx->host, ctx->port, NULL, 0);
+    if (!client) {
+        flb_plg_error(ctx->ins, "failed to create HTTP client");
+        flush_ret = FLB_ERROR;
+        goto done;
+    }
+
+    if (ctx->auth_token && flb_sds_len(ctx->auth_token) != 0) {
+        flb_http_add_header(client, "Authentication", 10, ctx->auth_token, strlen(ctx->auth_token));
+    }
+
+    flb_http_add_header(client, "Content-Type", 12, "application/json", 16);
+    flb_http_add_header(client, "User-Agent", 10, "Fluent-Bit", 10);
+
+    if (check_sw_under_test() == FLB_TRUE) {
+        tmp_ret = mock_oap_request(client, 200);
+    } else {
+        tmp_ret = flb_http_do(client, &sent_size);
+    }
+
+    if (tmp_ret == 0) {
+        flb_plg_debug(ctx->ins, "%s:%i, HTTP status=%i", ctx->host, ctx->port, client->resp.status);
+
+        if (client->resp.status < 200 || client->resp.status > 205) {
+            flush_ret = FLB_RETRY;
+        } else {
+            flush_ret = FLB_OK;
+        }
+    } else {
+        flb_plg_error(ctx->ins, "failed to flush buffer to %s:%i", ctx->host, ctx->port);
+        flush_ret = FLB_RETRY;
+    }
+
+done:
+    if (buf) {
+        flb_sds_destroy(buf);
+    }
+    if (client) {
+        flb_http_client_destroy(client);
+    }
+    if (conn) {
+        flb_upstream_conn_release(conn);
+    }
+
+    FLB_OUTPUT_RETURN(flush_ret);
+}
+
+static int cb_sw_exit(void *data, struct flb_config *config) {
+    struct flb_output_sw *ctx;
+    ctx = (struct flb_output_sw*)data;
+    if (!ctx) {
+        return 0;
+    }
+
+    if (ctx->host) {
+        flb_sds_destroy(ctx->host);
+    }
+    if (ctx->auth_token) {
+        flb_sds_destroy(ctx->auth_token);
+    }
+    if (ctx->svc_name) {
+        flb_sds_destroy(ctx->svc_name);
+    }
+    if (ctx->svc_inst_name) {
+        flb_sds_destroy(ctx->svc_inst_name);
+    }
+    if (ctx->u) {
+        flb_upstream_destroy(ctx->u);
+    }
+    if (ctx->http_scheme) {
+        flb_sds_destroy(ctx->http_scheme);
+    }
+    if (ctx->uri) {
+        flb_sds_destroy(ctx->uri);
+    }
+
+    flb_free(ctx);
+    return 0;
+}
+
```

Comment:
please implement a config map for your configuration, reference: https://github.com/fluent/fluent-bit/blob/master/plugins/out_http/http.c#L349

Reply:
implemented with config map.